### PR TITLE
Fix issue with @nogc breaking the signal API

### PR DESCRIPTION
**Pull request by Hackerpilot**
This is a regression from 2.065, and needs to make it into the next beta.

Context: http://forum.dlang.org/thread/myqafgvflwmttmqhhtti@forum.dlang.org

### CODE REVIEW

File: src/core/stdc/signal.d [hunk 1]
```diff
@@ -14,13 +14,14 @@ module core.stdc.signal;
...
-nothrow:
-@nogc:
 
 // this should be volatile
-alias int sig_atomic_t;
+alias sig_atomic_t = int;
+
+private alias sigfn_t = void function(int);
 
-private alias void function(int) sigfn_t;
```
Comment (Orvid) [inline src/core/stdc/signal.d]: This shouldn't be a private alias to begin with. This type is visible via the public API.

**Comment by Orvid** [comment]
Any updates on this?

**Comment by MartinNowak** [comment]
Someone should still fix it, fairly easy.

**Comment by MartinNowak** [comment]
Since it took so long. we can probably just leave things as they are.

**Comment by Orvid** [comment]
I think this is still needed. My comment about `sigfn_t` being `private` is still a problem.

**Comment by MartinNowak** [comment]
How about you make a pull then, should be trivial and I'm extremely short on time.

**Pull request closed without merging**